components/formats-gpl/src/loci/formats/in/VectraReader.java

```diff
@@ -538,6 +542,25 @@ else if (name.equals("ExposureTime")) {
         LOGGER.warn("Could not parse XML for channel {}", c);
         LOGGER.debug("", e);
       }
+
+      // set channel names based on "Name" and "Biomarker" attributes
+      // the "Biomarker" is preferred, in which case "Name" (if present) will be
+      // stored in the Channel.Fluor attribute
+      // if "Biomarker" is not present, then Channel.Name will be set to "Name"
+
+      boolean validBiomarker = biomarker != null && !biomarker.isEmpty();
+      boolean validName = storedChannelName != null && !storedChannelName.isEmpty();
+      for (int series=0; series<omeImageCount; series++) {
+        if (validBiomarker) {
+          store.setChannelName(biomarker, series, c);
+          if (validName) {
+            store.setChannelFluor(storedChannelName, series, c);
+          }
+        }
+        else {
+          store.setChannelName(storedChannelName, series, c);
```

Comment:
an empty string could be stored. Make check if it is a ``validName`` too.

Reply:
I intentionally didn't do that, so that the behavior would be unchanged in the common case where biomarker data is missing. If you're OK with the behavior changing in that case, I'll switch line 560 to `else if (validName)` (which will mean that `setChannelName` isn't guaranteed to be called for every channel).